src/types.ts

```diff
@@ -11,16 +11,19 @@ export namespace Serializer {
     metadata?: Metadata
   }
 
+  type TextPosition = number
+
+  export type TextRange = {
```

Comment:
Since the type's being recreated here and/or where it's being consumed is independent from the protobuf definition makes it so that tsc won't balk.

It'd imagine it's a matter of figuring out how tightly we want to couple against concrete protobuf types. However, it's kinda the point of using type-/codegen to ensure interface compliance using compiler infra.

<img width="997" alt="image" src="https://user-images.githubusercontent.com/250527/233386275-1814a0d6-dc3b-4017-9b27-d1637b39a23b.png">

Reply:
Yeah, the reason I didn't make gRPC types more explicit here is because I was going along with the existing type implementation, which is a remnant of the WASM module. I agree that it should all be based off of gRPC types instead.

How would you feel about splitting this into a separate ticket & PR? Since it affects more than just this part of the code and is otherwise largely independent of codelens support. @sourishkrout

Reply:
New ticket works. We might want to make it about removing the WASM serializer interface and standardizing on gRPC. Don't think we have needed to switch back to WASM at all thus far.